### ISSUE TITLE
Using lighthouse as a cargo dep fails with version mismatch

### ISSUE THREAD

**Issue by ralexstokes** [sigp/lighthouse#1690]
I'd like to be able to use various lighthouse components in my own Rust code.

One way to do this:

```toml
[dependencies]
lighthouse = { git = "https://github.com/sigp/lighthouse", tag = "v0.2.13" }
```

This currently fails (e.g. at the given tag) as there are some dependencies on `crates.io` that seem to be outdated.

Output of `cargo update` with a `Cargo.toml` with the above dep:
```bash
    Updating git repository `https://github.com/sigp/lighthouse`
    Updating crates.io index
    Updating git repository `https://github.com/sigp/blst.git`
    Updating git repository `https://github.com/sigp/milagro_bls`
error: failed to select a version for the requirement `tree_hash_derive = "^0.2.0"`
candidate versions found which didn't match: 0.1.0
location searched: crates.io index
required by package `types v0.2.0 (https://github.com/sigp/lighthouse?tag=v0.2.13#c0e76d2c)`
    ... which is depended on by `lighthouse v0.2.13 (https://github.com/sigp/lighthouse?tag=v0.2.13#c0e76d2c)`
```

**Comment by paulhauner** [comment]
Oh I think this might be due to patches:

https://github.com/sigp/lighthouse/blob/c0e76d2c15db499b6f901ac9e7e358bd0d4a8d89/Cargo.toml#L74-L81

I'm not familiar with how patches work in this scenario, but I'm sure there's a reasonable solution.

I agree it would be nice to do this, I found it useful when playing with the client formally known as Parity Ethereum.

**Comment by paulhauner** [comment]
I believe this has been resolved via #2472
